dashboard/client/dashboard.js

```diff
@@ -99,19 +112,53 @@ Template.dashboard.events({
       inputVars[key] = value;
     });
     Session.set('variableInputs', inputVars);
+  },
+  'click #addHeaderButton': function (event, template) {
+    var headerName = template.find('#newHeaderName').value;
+    var headerValue = template.find('#newHeaderValue').value;
+    headers =  ApiConfigs.findOne(activeConfigId.get()).headers;
+
+    if(_.contains(headers, headerName)) {
+      alert("Duplicate header"); //TODO: Replace with a reactive error
+    }
+    else {
+      template.find('#newHeaderName').value = "";
+      template.find('#newHeaderValue').value = "";
+
+      ApiConfigs.update(activeConfigId.get(), {$push: { headers: headerName }});
+      template.headerValues.set(headerName, headerValue);
+    }
+  },
+  'click #removeHeaderButton': function (event, template) {
+    ApiConfigs.update(activeConfigId.get(), {$pull: {headers: event.target.getAttribute('data-label')}} );
+  },
+  'keyup .headersName': function (event) {
+    // TODO update header in DB and in headerValus:
```

Comment:
I see the issue. You want to set "headers.${n}" where n is the index, so something like `ApiConfigs.update(activeConfigId.get(). {$set: { "headers.0": event.target.value }})` if the particular header is the first header. But we don't actually know the index. You could maybe iterate through activeConfig().headers until you match. 

Actually I think it might be better to add the index onto the field using a data-* attribute, then use it in here. I'll push a commit with that in a minute...